tests/worker/test_workflow.py

```diff
@@ -1141,6 +1152,139 @@ async def test_workflow_continue_as_new(client: Client):
         assert result[0] == handle.first_execution_run_id
 
 
+sa_prefix = "python_test_"
+
+
+def search_attrs_to_dict_with_type(attrs: SearchAttributes) -> Mapping[str, Any]:
+    return {
+        k: {
+            "type": type(vals[0]).__name__ if vals else "<unknown>",
+            "values": [str(v) if isinstance(v, datetime) else v for v in vals],
+        }
+        for k, vals in attrs.items()
+    }
+
+
+@workflow.defn
+class SearchAttributeWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        # Wait forever
+        await asyncio.Future()
+
+    @workflow.query
+    def get_search_attributes(self) -> Mapping[str, Mapping[str, Any]]:
+        return search_attrs_to_dict_with_type(workflow.info().search_attributes or {})
+
+    @workflow.signal
+    def do_search_attribute_update(self) -> None:
+        workflow.upsert_search_attributes(
+            {
+                f"{sa_prefix}text": ["text3"],
+                # We intentionally leave keyword off to confirm it still comes back
+                f"{sa_prefix}int": [123, 456],
+                # Empty list to confirm removed
+                f"{sa_prefix}double": [],
+                f"{sa_prefix}bool": [False],
+                f"{sa_prefix}datetime": [
+                    datetime(2003, 4, 5, 6, 7, 8, tzinfo=timezone(timedelta(hours=9)))
+                ],
+            }
+        )
+
+
+async def test_workflow_search_attributes(server: ExternalServer, client: Client):
+    if not server.supports_custom_search_attributes:
+        pytest.skip("Custom search attributes not supported")
+
+    # Add search attributes if not already present
+    async def search_attributes_present() -> bool:
+        resp = await client.service.get_search_attributes(GetSearchAttributesRequest())
+        return any(k for k in resp.keys.keys() if k.startswith(sa_prefix))
+
+    if not await search_attributes_present():
```

Comment:
```suggestion
    async def search_attributes_present() -> bool:
        resp = await client.service.get_search_attributes(GetSearchAttributesRequest())
        return any(k for k in resp.keys.keys() if k.startswith(sa_prefix))

    # Add search attributes if not already present
    if not await search_attributes_present():
```

Github mangles this, but, just moving the comment down after the function def. Found its current position a little confusing.

Reply:
Yeah, auto-formatter added spaces between my func def and my if statement. Before it was all one packed paragraph of code. Fixed.